cranelift-codegen/meta/src/cdsl/encodings.rs

```diff
@@ -86,11 +86,12 @@ impl EncodingBuilder {
                 }
 
                 // Add immediate value predicates
-                for (immediate_value, immediate_operand) in inst
-                    .immediate_values
-                    .iter()
-                    .zip(inst.inst.operands_in.iter().filter(|o| o.is_immediate()))
-                {
+                for (immediate_value, immediate_operand) in inst.immediate_values.iter().zip(
+                    inst.inst
+                        .operands_in
+                        .iter()
+                        .filter(|o| o.is_immediate_or_entityref()),
```

Comment:
I probably really should have used `is_pure_immediate` here but must have missed it so I used `is_immediate` instead; I think we can probably leave this as `is_immediate`.